File: NPB-CPP-cadna/cadna_c-3.1.11/examplesC/ex3_cad.cc
#include <iostream>
#include <cadna.h>

using namespace std;
int main()
{
  double_st amat[11][11];
  int i,j,k;
  double_st aux, det;

  cadna_init(-1);
  cout << "--------------------------------------------------------" << endl;
  cout << "|  Computation of the determinant of Hilbert's matrix  |" << endl; 
  cout << "|  using Gaussian elimination with CADNA               |" << endl;
  cout << "--------------------------------------------------------" << endl;

  for(i=1;i<=11;i++)
    for(j=1;j<=11;j++)
      amat[i-1][j-1] = 1./(double_st)(i+j-1);
  
  det = 1.;

  for(i=0;i<10;i++){
    cout << "Pivot number " << i << " = " << amat[i][i] << endl;
    det = det*amat[i][i];
    aux = 1./amat[i][i];
    for(j=i+1;j<11;j++)
      amat[i][j] = amat[i][j]*aux;
    
    for(j=i+1;j<11;j++){
      aux = amat[j][i];
      for(k=i+1;k<11;k++)
	amat[j][k] = amat[j][k] - aux*amat[i][k];
    }
  }
  cout << "Pivot number " << i << " = "  << amat[i][i] << endl;
  det = det*amat[i][i];
  cout << "Determinant      = " << det << endl;
  cout << "Remark : only the first two digits are exact : 3.0e-65"<< endl; 
  cadna_end();
	  
}

File: NPB-CPP-cadna/cadna_c-3.1.11/examplesFortran/ex3_cad.f90
      program ex3_cad
      use cadna
      implicit type (double_st) (a-h,o-z)
      dimension amat(11,11)
      call cadna_init(-1)
      !
      print *,'Computation of the determinant of Hilbert matrix'
      print *,'using Gaussian elimination with CADNA:'
      !
      do i=1,11
        do j=1,11
          amat(i,j) = 1.d0/(i+j-1)
        enddo
      enddo
      det = 1.d0
!
      do i=1,10
        write(*,'(a13,i2,a4,a23)')'Pivot number ',i,'  = ',&
     &                             str(amat(i,i))
        det = det*amat(i,i)
        aux = 1.d0/amat(i,i)
        do j=i+1,11
          amat(i,j) = amat(i,j)*aux
        enddo
!
        do j=i+1,11
          aux = amat(j,i)
          do k=i+1,11
            amat(j,k) = amat(j,k) - aux*amat(i,k)
          enddo
        enddo
!
      enddo
!
      write(*,'(a13,i2,a4,a23)')'Pivot number ',i,'  = ',str(amat(i,i))
      det = det*amat(i,i)
      write(*,'(a19,a23)')'Determinant      = ',str(det)
      call cadna_end()
!
      end
